service/tbc/indexer.go

```diff
@@ -455,6 +455,13 @@ func (c *indexerCommon) parseBlocks(ctx context.Context, endHash *chainhash.Hash
 			break
 		}
 
+		// Check for sigterm.
+		select {
+		case <-ctx.Done():
+			return 0, last, ctx.Err()
+		default:
+		}
```

Comment:
This can be simplified as:

```suggestion
		if err := ctx.Err(); err != nil {
			return 0, last, err
		}
```